From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Thu, 4 Apr 2019 19:34:24 +0200
Subject: [PATCH] {MAYBE UPS} mm/kasan: initialize tag

clang reports:

    mm/kasan/common.c:492:40: error: variable 'tag' is uninitialized
    when used here [-Werror,-Wuninitialized]
            kasan_unpoison_shadow(set_tag(object, tag), size);
                                                  ^~~
    mm/kasan/kasan.h:178:65: note: expanded from macro 'set_tag'
    #define set_tag(addr, tag)      ((void *)arch_kasan_set_tag((addr), (tag)))
                                                                         ^~~
    mm/kasan/common.c:475:8: note: initialize the variable 'tag' to
    silence this warning
            u8 tag;
                  ^
                   = '\0'

... but a comment explains that the parameter is not used.
---
 mm/kasan/common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/kasan/common.c b/mm/kasan/common.c
index 36afcf64e016..4c5af68f2a8b 100644
--- a/mm/kasan/common.c
+++ b/mm/kasan/common.c
@@ -464,7 +464,7 @@ static void *__kasan_kmalloc(struct kmem_cache *cache, const void *object,
 {
 	unsigned long redzone_start;
 	unsigned long redzone_end;
-	u8 tag;
+	u8 tag = 0;
 
 	if (gfpflags_allow_blocking(flags))
 		quarantine_reduce();
-- 
